dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -421,4 +457,41 @@ def find_counterfactuals(self, query_instance, desired_range, desired_class, sto
             print('Diverse Counterfactuals found! total time taken: %02d' %
                   m, 'min %02d' % s, 'sec')
 
-        return query_instance
\ No newline at end of file
+        return query_instance_df
+
+    def label_encode(self, input_instance):
+        for column in self.data_interface.categorical_feature_names:
+            input_instance[column] = self.labelencoder[column].transform(input_instance[column])
+        return input_instance
+
+    def label_decode(self, labelled_input):
```

Comment:
def from_label(self, data):
        """Transforms label encoded data back to categorical values"""
        out = data.copy()
        if isinstance(data, pd.DataFrame) or isinstance(data, dict):
            for column in self.categorical_feature_names:
                out[column] = self.labelencoder[column].inverse_transform(out[column].round().astype(int).tolist())
            return out
        elif isinstance(data, list):
            for c in self.categorical_feature_indexes:
                out[c] = self.labelencoder[self.feature_names[c]].inverse_transform([round(out[c])])[0]
            return out